### PR TITLE
Check if project has the same repository id with issue when assign project to issue

### PR DESCRIPTION
**Pull request by lunny**

### CODE REVIEW

File: models/issues/milestone.go [hunk 1]
```diff
@@ -124,6 +124,11 @@ func NewMilestone(m *Milestone) (err error) {
 	return committer.Commit()
 }
 
+// HasMilestoneByRepoID returns if the  milestone exists in the repository.
```
Comment (KN4CK3R) [inline models/issues/milestone.go]:
```suggestion
// HasMilestoneByRepoID returns if the milestone exists in the repository.
```

File: services/issue/milestone.go [hunk 2]
```diff
@@ -15,6 +15,17 @@ import (
 )
 
 func changeMilestoneAssign(ctx context.Context, doer *user_model.User, issue *issues_model.Issue, oldMilestoneID int64) error {
+	// Only check if milestone exists if we don't remove it.
+	if issue.MilestoneID > 0 {
+		has, err := issues_model.HasMilestoneByRepoID(ctx, issue.RepoID, issue.MilestoneID)
+		if err != nil {
+			return fmt.Errorf("GetMilestoneByRepoID: %v", err)
```
Comment (KN4CK3R) [inline services/issue/milestone.go]:
```suggestion
			return fmt.Errorf("HasMilestoneByRepoID: %v", err)
```

File: services/issue/milestone.go [hunk 3]
```diff
@@ -15,6 +15,17 @@ import (
...
+		}
+		if !has {
+			return fmt.Errorf("GetMilestoneByRepoID: issue doesn't exist")
```
Comment (KN4CK3R) [inline services/issue/milestone.go]:
```suggestion
			return fmt.Errorf("HasMilestoneByRepoID: issue doesn't exist")
```

**Review by lafriks — APPROVED** [review]

**Review by a1012112796 — APPROVED** [review]

**Comment by lunny** [comment]
make L-G-T-M work

**Pull request merged**